percona_pg_telemetry.c

```diff
@@ -891,6 +892,8 @@ percona_pg_telemetry_main(Datum main_arg)
 	/* This is the context that we will allocate our data in */
 	pt_cxt = AllocSetContextCreate(TopMemoryContext, "Percona Telemetry Context", ALLOCSET_DEFAULT_SIZES);
 
+	pg_version = GetConfigOptionByName("server_version", NULL, true);
```

Comment:
I don't have a strong opinion here but I am leaning towards calling the code with `missing_ok = false` or add an `Assert(pg_version != NULL).` just to make it very clear that it can never be NULL.

Reply:
I checked that rest of the code works correctly if `pg_version` is NULL, nothing crashes, etc. And I'm kinda scared about making such hard assertions, mostly because I'm not completely sure if this function can fail for this option in any case. From comments in sources It looks like it `always` there, but I wasn't 100% sure.

If you think that using assertion for that case is fine, then ok.

Reply:
Pushed change